### PR TITLE
feat: create date of birth pattern tckt-361

### PR DESCRIPTION
**Pull request by kalasgarov**
Created  Date of Birth Pattern for Forms Users.
Created  Date of Birth Pattern Edit for Form Builders.
Added unit tests for Date of Birth Pattern.

Refactored and added value aggregation hook to handle structured and single values in form validation

### CODE REVIEW

File: packages/forms/src/pattern.ts [hunk 1]
```diff
@@ -133,6 +133,27 @@ export const validatePattern = (
   return r.success(parseResult.data);
 };
 
+const aggregateValuesByPrefix = (
+  values: Record<string, string>
+): Record<string, any> => {
+  const aggregatedValues: Record<string, any> = {};
+
+  for (const [key, value] of Object.entries(values)) {
+    const [prefix, suffix] = key.split('.');
```
Comment (danielnaab) [inline packages/forms/src/pattern.ts:142]: Glad you figured out what was going on here!

I think we may want to do something like this, to deal with more deeply nested objects, such as `abc.def.0.xyz`:

```suggestion
    const [prefix, ...suffix] = key.split('.');
```

Then we'd need to handle the suffix as an array, perhaps by making this a recursive function...

There are also several libraries that do this, which may be easier. This one is already in our lock file:

https://www.npmjs.com/package/set-value

And this one is also heavily used:

https://www.npmjs.com/package/dot-prop

File: packages/forms/src/patterns/date-of-birth/date-of-birth.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    .regex(/^\d{1,2}$/, 'Invalid day format')
+    .optional();
+  const monthSchema = z
+    .string()
+    .regex(/^(0[1-9]|1[0-2])$/, 'Invalid month format')
+    .optional();
+  const yearSchema = z
+    .string()
+    .regex(/^\d{4}$/, 'Invalid year format')
+    .optional();
```
Comment (danielnaab) [inline packages/forms/src/patterns/date-of-birth/date-of-birth.ts:39]: These schemas look correct, but I think we may want them to be coerced into numbers. @ethangardner thoughts?

**Review by danielnaab — CHANGES_REQUESTED** [review]
Great work tracking down the nested keys issue! Would it be possible to update it to support nesting deeper than one level?

File: packages/forms/src/pattern.ts (hunk 1 above)
Comment (kalasgarov) [reply packages/forms/src/pattern.ts:142]: Good point! Updating it to set-value.

Comment (kalasgarov) [reply packages/forms/src/pattern.ts]: Updated!

**Review by danielnaab — APPROVED** [review]
👍

File: packages/forms/src/patterns/date-of-birth/date-of-birth.ts (hunk 2 above)
Comment (kalasgarov) [reply packages/forms/src/patterns/date-of-birth/date-of-birth.ts:39]: I am going to merge the current PR to unblock my next work, but in some point if want to update anything here I can raise a separate PR for it.

**Pull request merged**